migration/migrator/migrations/course/20190611034600_add_instructor_solution_directory.py

```diff
@@ -0,0 +1,30 @@
+"""Migration for a given Submitty course database."""
+
+import os
+import grp
+import pwd
+from pathlib import Path
+import shutil
+
+def up(config, database, semester, course):
+
+    course_dir = Path(config.submitty['submitty_data_dir'], 'courses', semester, course)
+    solution_dir = Path(course_dir, 'instructor_solution')
+
+    # get course group
+    stat_info = os.stat(str(course_dir))
+    course_group_id = stat_info.st_gid
+    uid = stat_info.st_uid
+    course_group = grp.getgrgid(course_group_id)[0]
+    user = pwd.getpwuid(uid)[0]
+
+    if not os.path.exists(solution_dir):
+        os.mkdir(solution_dir)
+        shutil.chown(solution_dir, user=user, group=course_group)
+        os.system("chmod u+rwx {0}".format(str(solution_dir)))
+        os.system("chmod g+rwx {0}".format(str(solution_dir)))
```

Comment:
just in case of bad umask, lets set other -rwx
os.system("chmod o-rwx {0}".format(str(solution_dir)))

and also set the group sgid bit so subdirectories inherit the group
os.system("chmod g+s {0}".format(str(solution_dir)))